### PR TITLE
refactor: standardize using formatAddress PE-6982

### PR DESCRIPTION
**Pull request by fedellen**

### CODE REVIEW

File: src/main.lua [hunk 1]
```diff
@@ -321,10 +321,14 @@ end, function(msg)
 		lastKnownWithdrawSupply = LastKnownWithdrawSupply,
 		lastKnownTotalSupply = lastKnownTotalTokenSupply(),
 	}
+
+	-- Stash the formatted from address
+	msg.formattedFrom = utils.formatAddress(msg.From)
```
Comment (arielmelendez) [inline src/main.lua:326]: I envisioned this as just replacing the `from` field altogether. And additionally formatting other fields if they exist, e.g.: 
```
msg.tags.Recipient = msg.tags.Recipient and utils.formatAddress(msg.tags.Recipient) or nil
```
and so on for any other address values. I haven't fully thought through whether doing that "destructively" is a bad idea. If we don't want to destroy the incoming value, we can always create more fields as you've done here.

Comment (fedellen) [reply src/main.lua]: 34bd2b9

Comment (fedellen) [reply src/main.lua]: image

File: src/main.lua [hunk 2]
```diff
@@ -964,16 +964,15 @@ addEventingHandler(ActionMap.JoinNetwork, utils.hasMatchingTag("Action", ActionM
 	addSupplyData(msg.ioEvent)
 
 	ao.send({
-		Target = fromAddress,
+		Target = msg.FromAddress,
```
Comment (fedellen) [inline src/main.lua:967]:
```suggestion
		Target = msg.From,
```

File: src/main.lua [hunk 3]
```diff
@@ -1160,21 +1156,20 @@ addEventingHandler(ActionMap.DelegateStake, utils.hasMatchingTag("Action", Actio
 
 	ao.send({
 		Target = msg.From,
-		Tags = { Action = ActionMap.DelegateStake .. "-Notice", Gateway = msg.Tags.Target },
+		Tags = { Action = ActionMap.DelegateStake .. "-Notice", Gateway = msg.Target },
```
Comment (fedellen) [inline src/main.lua:1159]:
```suggestion
		Tags = { Action = ActionMap.DelegateStake .. "-Notice", Gateway = msg.Tags.Target },
```

File: src/main.lua [hunk 4]
```diff
@@ -370,6 +370,19 @@ end, function(msg)
 		lastKnownRequestSupply = LastKnownPnpRequestSupply,
 		lastKnownTotalSupply = lastKnownTotalTokenSupply(),
 	}
+
+	-- Format all incoming addresses
+	msg.From = utils.formatAddress(msg.From)
+	msg.Tags.Recipient = msg.Tags.Recipient and utils.formatAddress(msg.Tags.Recipient) or nil
```
Comment (arielmelendez) [inline src/main.lua:376]: Nit: this could be done in a for loop over an array of known tags.

File: src/main.lua [hunk 5]
```diff
@@ -1254,27 +1248,20 @@ addEventingHandler(
...
+			quantity and utils.isInteger(quantity) and quantity > constants.minimumWithdrawalAmount,
 			"Invalid quantity. Must be integer greater than " .. constants.minimumWithdrawalAmount
 		)
 
-		local result = gar.decreaseDelegateStake(target, from, quantity, timestamp, messageId, instantWithdraw)
-
+		local result = gar.decreaseDelegateStake(target, msg.From, quantity, timestamp, msg.Id, instantWithdraw)
 		local decreaseDelegateStakeResult = {
 			gateway = result and result.gateway or {},
-			penaltyRate = result and result.penaltyRate or 0,
```
Comment (arielmelendez) [inline src/main.lua:1277]: Is this relevant to the PR?

File: src/main.lua [hunk 6]
```diff
@@ -370,6 +370,19 @@ end, function(msg)
...
+	msg.Tags.Recipient = msg.Tags.Recipient and utils.formatAddress(msg.Tags.Recipient) or nil
+	msg.Tags.Initiator = msg.Tags.Initiator and utils.formatAddress(msg.Tags.Initiator) or nil
+	msg.Tags.Target = msg.Tags.Target and utils.formatAddress(msg.Tags.Target) or nil
+	msg.Tags.Source = msg.Tags.Source and utils.formatAddress(msg.Tags.Source) or nil
+	msg.Tags.Address = msg.Tags.Address and utils.formatAddress(msg.Tags.Address) or nil
+	msg.Tags["Vault-Id"] = msg.Tags["Vault-Id"] and utils.formatAddress(msg.Tags["Vault-Id"]) or nil
+	msg.Tags["Process-Id"] = msg.Tags["Process-Id"] and utils.formatAddress(msg.Tags["Process-Id"]) or nil
+	msg.Tags["Observer-Address"] = msg.Tags["Observer-Address"] and utils.formatAddress(msg.Tags["Observer-Address"])
+		or nil
+
```
Comment (dtfiedler) [inline src/main.lua:385]: nit add Quantity and convert it to number

File: src/main.lua [hunk 7]
```diff
@@ -1197,7 +1192,7 @@ addEventingHandler(ActionMap.CancelWithdrawal, utils.hasMatchingTag("Action", Ac
 	end
 
 	ao.send({
-		Target = msg.From,
+		Target = msg.FromAddress,
```
Comment (dtfiedler) [inline src/main.lua:1195]:
```suggestion
		Target = msg.From,
```

File: src/main.lua (hunk 5 above)
Comment (fedellen) [reply src/main.lua:1277]: [20e7aa0](https://github.com/ar-io/ar-io-network-process/pull/183/commits/20e7aa08098a40ba6ca75d3f28c629d04c9d8c45)

File: src/main.lua (hunk 6 above)
Comment (fedellen) [reply src/main.lua:392]: [119a06f](https://github.com/ar-io/ar-io-network-process/pull/183/commits/119a06f78258d77ff51bbf0e5f9e62360c94d207)

File: src/main.lua (hunk 4 above)
Comment (fedellen) [reply src/main.lua]: [a4ad6cd](https://github.com/ar-io/ar-io-network-process/pull/183/commits/a4ad6cdd20fbc2816860b332288243d6cf78d500)

**Review by dtfiedler — APPROVED** [review]
